theano/tensor/var.py

```diff
@@ -768,8 +768,7 @@ def choose(self, a, choices, out=None, mode='raise'):
         from.
 
         """
-        return theano.tensor.basic.choose(self, a, choices, out=None,
-                                          mode='raise')
+        return theano.tensor.basic.choose(a, choices, out=None, mode='raise')
```

Comment:
yes those signatures were confusing, this makes much more sense. Updating.

As for testing all I can see is `class T_Choose(utt.InferShapeTester)` in `/tensor/tests/test_basic` seems like it should have covered this code.